tests/transformation/test_fixedpt_quantize.py

```diff
@@ -0,0 +1,496 @@
+    for tname in test_details["quant_dict"].keys():
+        tdtype = DataType[test_details["quant_dict"][tname]]
+        tdata = model.get_initializer(tname)
+
+        # Check if the tensor exists in the graph
+        assert tdata is not None
+
+        # Check if all values of the tensor are allowed with the target datatype
+        assert tdtype.allowed(tdata).all()
+
+        # Check if the maximum error is within the LSB bound of the datatype
+        allowed_max_error = tdtype.scale_factor()
+        # The bound reduces by a factor of 2 if the mode is "ROUND"
+        if test_details["rounding_mode"] == "ROUND":
+            allowed_max_error /= 2
+        assert fxp_transform.max_err[tname] <= allowed_max_error
+
+fixedpt_details = {
+    "FINN-CNV_W2A2_round_0": {
+        "test_model": "FINN-CNV_W2A2",
+        "dtype": "FIXED<8,3>",
+        "rounding_mode": "ROUND",
+        "quant_tensors": [
+            "Mul_0_param0",
+            "Mul_1_param0",
+            "Add_0_param0"
+        ]
+    },
+    "FINN-CNV_W2A2_floor_0": {
+        "test_model": "FINN-CNV_W2A2",
+        "dtype": "FIXED<8,3>",
+        "rounding_mode": "FLOOR",
+        "quant_tensors": [
+            "Mul_0_param0",
+            "Mul_1_param0",
+            "Add_0_param0"
+        ]
+    },
+    "FINN-CNV_W2A2_round_1": {
+        "test_model": "FINN-CNV_W2A2",
+        "dtype": "FIXED<4,3>",
+        "rounding_mode": "ROUND",
+        "quant_tensors": [
+            "Mul_0_param0",
+            "Mul_1_param0",
+            "Add_0_param0"
+        ]
+    },
+    "FINN-CNV_W2A2_floor_1": {
+        "test_model": "FINN-CNV_W2A2",
+        "dtype": "FIXED<4,3>",
+        "rounding_mode": "FLOOR",
+        "quant_tensors": [
+            "Mul_0_param0",
+            "Mul_1_param0",
+            "Add_0_param0"
+        ]
+    },
+    "FINN-CNV_W2A2_round_2": {
+        "test_model": "FINN-CNV_W2A2",
+        "dtype": "FIXED<12,3>",
+        "rounding_mode": "ROUND",
+        "quant_tensors": [
+            "Mul_0_param0",
+            "Mul_1_param0",
+            "Add_0_param0"
+        ]
+    },
+    "FINN-CNV_W2A2_floor_2": {
+        "test_model": "FINN-CNV_W2A2",
+        "dtype": "FIXED<12,3>",
+        "rounding_mode": "FLOOR",
+        "quant_tensors": [
+            "Mul_0_param0",
+            "Mul_1_param0",
+            "Add_0_param0"
+        ]
+    }
+}
+
+
+@pytest.mark.parametrize("test_case", fixedpt_details.keys())
+def test_fixedpt_quantize(test_case):
+    test_details = fixedpt_details[test_case]
+    dl_file = download_model(test_model=test_details["test_model"])
+    assert os.path.isfile(dl_file)
+    model = ModelWrapper(dl_file)
+    model = cleanup_model(model)
+
+    tdtype = test_details["dtype"]
+    fxp_transform = FixedPointQuantizeParams(tdtype, rounding_mode=test_details["rounding_mode"])
+    tdtype = DataType[tdtype]
+    model = model.transform(fxp_transform)
+    model = cleanup_model(model)
+    os.unlink(dl_file)
```

Comment:
minor: suggest to remove downloaded files at the end of the test. in case the test fails, this helps the debug a little bit